README.md

```diff
@@ -31,7 +33,22 @@ globParent('path/**/*'); // 'path'
 globParent('path/foo/bar.js'); // 'path/foo'
 globParent('path/foo/'); // 'path/foo'
 globParent('path/foo'); // 'path' (see issue #3 for details)
+```
+
+## Escaping
+
+The following characters have special significance in glob patterns and must be escaped if you want them to be treated as regular path characters:
 
+- `?` (question mark)
+- `*` (star)
+- `[` (opening bracket)
+- `]` (closing bracket)
+
+**Example**
+
+```js
+globParent('foo/[bar]/') // 'foo'
+globParent('foo/\\[bar]/') // 'foo/\\[bar]'
```

Comment:
Jon and I were just talking about a method that would replace escaped glob characters with a heuristic, then run dirname, then restore the characters again. We can implement this in `path-dirname` and allow an option to be passed to run the logic for handling globs, but otherwise it can act as a polyfill to make it useful to other libraries.